roofit/doc/developers/roofit_ad.md

```diff
@@ -92,22 +108,67 @@ On a high level, this *code generation* is implemented as follows:
    original computation graph, with the added benefit that it can be queried for
    the gradient.
 
-In summary, the important ingredient to enable AD in RooFit is to support the **C++ code generation** from RooFit classes.
+In summary, the important ingredient to enable AD in RooFit is to support the 
+**C++ code generation** from RooFit classes.
+
+# Steps to enable Code Generation in RooFit classes
+
+There are multiple code generation approaches that can be followed to add Code
+ Generation support in RooFit classes. 
+
+**Approach 1:** For very simple cases like `RooRatio`, it may be preferable to
+ write the entire code in a single string.
+
+**Approach 2:** Another approach could be to extract free functions in a 
+separate header file. 
+Since Clad prefers the code for models to be within a single translation unit,
+ in many classes, this has been implemented by moving the computational 
+aspects of the RooFit class
+to free functions in a single header file named [EvaluateFuncs] (and/or 
+[AnalyticalIntegrals], where relevant). This approach enables easier debugging
+ (e.g., you can standalone-compile the generated code with just a few header 
+files copied outside ROOT)
+
+*Refactoring* It is important to refactor the code such that:
+
+- the footprint of the generated code is minimized by referring to existing 
+functions with the definition known by interpreter (i.e., they are in public 
+header files).
+
+- to reuse common code, both in the generated code, and in the existing 
+evaluate() method (meaning that the refactoring of `evaluate()` is optional, 
```

Comment:
```suggestion
RooAbsReal::evaluate() method (meaning that the refactoring of `evaluate()` is optional, 
```
At least the first time any method is introduced in this document, I would use the full function name including the class so that Doxygen generates a clickable link.